src/app/core/store/user/user.effects.ts

```diff
@@ -256,6 +259,39 @@ export class UserEffects {
     )
   );
 
+  @Effect()
+  loadUserPaymentMethods$ = this.actions$.pipe(
+    ofType<userActions.LoadUserPaymentMethods>(userActions.UserActionTypes.LoadUserPaymentMethods),
+    withLatestFrom(this.store$.pipe(select(getLoggedInCustomer))),
+    filter(([, customer]) => !!customer),
+    concatMap(([, customer]) =>
+      this.paymentService.getUserPaymentMethods(customer).pipe(
+        map(result => new userActions.LoadUserPaymentMethodsSuccess({ paymentMethods: result })),
+        mapErrorToAction(userActions.LoadUserPaymentMethodsFail)
+      )
+    )
+  );
+
+  @Effect()
+  deleteUserPayment$ = this.actions$.pipe(
+    ofType<userActions.DeleteUserPayment>(userActions.UserActionTypes.DeleteUserPayment),
+    mapToPayloadProperty('id'),
+    withLatestFrom(this.store$.pipe(select(getLoggedInCustomer))),
+    filter(([, customer]) => !!customer),
+    concatMap(([id, customer]) =>
+      this.paymentService.deleteUserPaymentInstrument(customer.customerNo, id).pipe(
+        concatMapTo([
+          new userActions.DeleteUserPaymentSuccess(),
+          new userActions.LoadUserPaymentMethods(),
+          new SuccessMessage({
+            message: 'account.payment.payment_deleted.message',
```

Comment:
I changed the deletion message into  "This payment instrument has been deleted from your account."
I also renamed DeleteUserPayment actions into DeleteUserPaymentInstrument. The names for the basket actions (xxxBasketPayment) are ok, because they create/delete etc. an instrument and a payment as well